### PR TITLE
docs: add disk encryption guide

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: https://github.com/talos-systems/talos/issues/3175

Describe usage tips, caveats, flow.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

File: website/content/docs/v0.9/Guides/disk-encryption.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+---
+title: "Disk Encryption"
+description: "This page contains guides related kernel level disk encryption"
+---
+
+LUKS2 disk encryption is supported by the Linux kernel modules and `cryptsetup` utility.
+
+Sequencer runs additional steps when encryption is enabled.
```
Comment (smira) [inline website/content/docs/v0.9/Guides/disk-encryption.md:8]: I think `sequencer` is something internal to Talos, we never published it in the docs.

File: website/content/docs/v0.9/Guides/disk-encryption.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+---
+title: "Disk Encryption"
+description: "This page contains guides related kernel level disk encryption"
```
Comment (smira) [inline website/content/docs/v0.9/Guides/disk-encryption.md:3]: Guide on enabling system  disk encryption

?

File: website/content/docs/v0.9/Guides/disk-encryption.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+---
+
```
Comment (smira) [inline website/content/docs/v0.9/Guides/disk-encryption.md:5]: I think we should start with the overview - what gets encrypted, what is not encrypted and why

File: website/content/docs/v0.9/Guides/disk-encryption.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+- nodeID which is generated using the node UUID and the partition label (note that if the node UUID is not really random it will fail the entropy check).
+- static which you define right in the configuration.
+
+> Note: Use static keys only if your STATE partition is encrypted and only for the EPHEMERAL partition.
+> Otherwise the passphrase will be unencrypted.
+
+Take a note that keys order does not play any role on which key slot is used.
+Every key must always have a slot defined.
+
+### Keys Rotation
```
Comment (smira) [inline website/content/docs/v0.9/Guides/disk-encryption.md:78]: `Key Rotation` ?

File: website/content/docs/v0.9/Guides/disk-encryption.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+- before mounting the STATE partition, load encryption configs either from the machine config or from the META partition,
+  the machine config is always preferred over the META one.
+- before mounting the STATE partition, format and encrypt it, but only if it is empty and has no filesystem.
+
+If the disk encryption is enabled for the EPHEMERAL partition, the sequencer will:
+
+- get the encryption config from the machine config.
+- before mounting the EPHEMERAL partition, encrypt and format it, but only if it is empty and has not filesystem.
+
+## Configuration
```
Comment (smira) [inline website/content/docs/v0.9/Guides/disk-encryption.md:21]: we should talk about encryption key providers?

File: website/content/docs/v0.9/Guides/disk-encryption.md (hunk 1 above)
Comment (Unix4ever) [reply website/content/docs/v0.9/Guides/disk-encryption.md:8]: Will rename it to system

File: website/content/docs/v0.9/Guides/disk-encryption.md (hunk 2 above)
Comment (Unix4ever) [reply website/content/docs/v0.9/Guides/disk-encryption.md:3]: uuh, yeah, I messed up here

File: website/content/docs/v0.9/Guides/disk-encryption.md (hunk 5 above)
Comment (Unix4ever) [reply website/content/docs/v0.9/Guides/disk-encryption.md:21]: We have it there

File: website/content/docs/v0.9/Guides/disk-encryption.md (hunk 2 above)
Comment (Unix4ever) [reply website/content/docs/v0.9/Guides/disk-encryption.md:3]: but at the same time it's not only about enabling the disk encryption, but contains other use cases.

**Pull request merged**